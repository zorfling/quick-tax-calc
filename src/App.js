import React, { Component, Fragment } from 'react';
import styled from 'styled-components';
import mexp from 'math-expression-evaluator';

const Container = styled.div`
  align-items: center;
  display: flex;
  height: 100vh;
  justify-content: center;
  line-height: 150%;
`;

const Row = styled.div`
  display: flex;
`;

const Label = styled.div`
  font-weight: bold;
`;

const Value = styled.div`
  margin-left: auto;
`;

class App extends Component {
  state = {
    income: 90000,
    parsedIncome: 90000,
    rememberedFortnightlyAfterTax: null
  };

  calculateTax() {
    const { parsedIncome } = this.state;

    if (parsedIncome > 180000) {
      return 54232 + (parsedIncome - 180000) * 0.45;
    }

    if (parsedIncome > 87000 && parsedIncome < 180001) {
      return 19822 + (parsedIncome - 87000) * 0.37;
    }

    if (parsedIncome > 37000 && parsedIncome < 87001) {
      return 3572 + (parsedIncome - 37000) * 0.325;
    }

    if (parsedIncome > 18200 && parsedIncome < 37001) {
      return (parsedIncome - 18200) * 0.19;
    }

    return 0;
  }

  handleIncomeChange = evt => {
    this.parseAndSetIncome(evt.target.value);
  };

  parseAndSetIncome = newIncome => {
    let parsedIncome;
    try {
      parsedIncome = mexp.eval(newIncome);
    } catch (e) {
      this.setState({
        income: newIncome
      });
      return;
    }

    this.setState({
      income: newIncome,
      parsedIncome
    });
  };

  handleRemember = () => {
    this.setState(prevState => ({
      rememberedFortnightlyAfterTax:
        (prevState.parsedIncome - this.calculateTax()) / 26
    }));
  };

  handleIncomeTemplate = template => () => {
    this.parseAndSetIncome(template);
  };

  formatCurrency(amount) {
    return Number(amount).toLocaleString(
      'en-AU',
      {
        currency: 'AUD',
        style: 'currency'
      }
    );
  }

  render() {
    const { income, parsedIncome, rememberedFortnightlyAfterTax } = this.state;
    const annualAfterTax = parsedIncome - this.calculateTax();
    return (
      <Container>
        <div>
          <h1>Quick Tax Calc</h1>
          <p>
            This is a quick calculator for the{' '}
            <a href="https://www.ato.gov.au/Rates/Individual-income-tax-rates/">
              ATO Australian Individual Tax Rates
            </a>
          </p>
          <p>
            Type your annual pre-tax amount in the income section and see your
            after tax annual and fortnightly amounts.
          </p>
          <p>
            You can use the memory button to remember a given fortnightly amount
            for comparison.
          </p>
          <p>
            The income section also allows for formulae so you can do things
            like:
          </p>
          <ul>
            <li>
              How might a 10% raise affect my income?
              <button onClick={this.handleIncomeTemplate(`90000 * 1.1`)}>
                Try it
              </button>
            </li>
            <li>
              What if I get an extra job paying 500 a week?
              <button onClick={this.handleIncomeTemplate(`90000 + 500 * 52`)}>
                Try it
              </button>
            </li>
            <li>
              If I go 3 days a week, how much will I get paid?
              <button onClick={this.handleIncomeTemplate(`90000 * 3 / 5`)}>
                Try it
              </button>
            </li>
          </ul>
          <Row style={{ marginBottom: '2rem' }}>
            <Label>Income:</Label>{' '}
            <Value>
              <input
                type="text"
                value={income}
                onChange={this.handleIncomeChange}
                style={{
                  fontSize: '1.4em',
                  textAlign: 'right',
                  border: 'none'
                }}
              />
              <small style={{ display: 'block', textAlign: 'right' }}>
                Accepts formulas - eg 90000 * 1.1
              </small>
            </Value>
          </Row>
          <Row>
            <Label>Income:</Label>
            <Value>{this.formatCurrency(parsedIncome)}</Value>
          </Row>
          <Row>
            <Label>Tax:</Label>
            <Value>{this.formatCurrency(this.calculateTax())}</Value>
          </Row>
          <Row style={{ marginBottom: '2rem' }}>
            <Label>Annual Less Tax:</Label>
            <Value>{this.formatCurrency(annualAfterTax)}</Value>
          </Row>
          <Row>
            <Label>Fortnightly:</Label>
            <Value>{this.formatCurrency(parsedIncome / 26)}</Value>
          </Row>
          <Row>
            <Label>Fortnightly Tax:</Label>
            <Value>{this.formatCurrency(this.calculateTax() / 26)}</Value>
          </Row>
          <Row style={{ marginTop: '2rem' }}>
            <Label>Fortnightly Less Tax:</Label>
            <Value style={{ fontSize: '1.5em', marginRight: '-1.1em' }}>
              {this.formatCurrency(annualAfterTax / 26)}
              <button onClick={this.handleRemember}>M</button>
            </Value>
          </Row>
          {rememberedFortnightlyAfterTax ? (
            <Fragment>
              <Row>
                <Label>Remembered:</Label>
                <Value style={{ fontSize: '1.5em', lineHeight: '1.5em' }}>
                  {this.formatCurrency(rememberedFortnightlyAfterTax)}
                </Value>
              </Row>
              <Row>
                <Label>Difference:</Label>
                <Value style={{ fontSize: '1.5em' }}>
                  {this.formatCurrency(
                    annualAfterTax / 26 - rememberedFortnightlyAfterTax
                  )}
                </Value>
              </Row>
            </Fragment>
          ) : null}
        </div>
      </Container>
    );
  }
}

export default App;
