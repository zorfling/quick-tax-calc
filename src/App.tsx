import React, { Component, Fragment, ChangeEvent } from 'react';
import styled from 'styled-components';
import mexp from 'math-expression-evaluator';
import calculateTax from './calculateTax';

const Container = styled.div`
  align-items: center;
  display: flex;
  min-height: 100vh;
  justify-content: center;
  line-height: 150%;
`;

const Card = styled.div`
  display: flex;
  flex-direction: column;
  box-shadow: 0px 2px 6px 2px hsl(0, 0%, 0%, 0.2);
  padding: 1rem 3rem;

  @media screen and (max-width: 575px) {
    > p,
    > ul,
    > h2 {
      order: 1;
    }

    p:nth-of-type(1) {
      order: 0;
    }
  }
`;

const Row = styled.div`
  display: flex;
  justify-content: space-between;
`;

const Label = styled.div`
  font-weight: bold;
`;

const FormLabel = styled.label`
  font-weight: bold;
`;

const Value = styled.div`
  margin-right: 2.1rem;
`;

interface AppState {
  income: string;
  parsedIncome: number;
  rememberedFortnightlyAfterTax: number;
}

class App extends Component<{}, AppState> {
  state = {
    income: '90000',
    parsedIncome: 90000,
    rememberedFortnightlyAfterTax: -1
  };

  calculateTax() {
    const { parsedIncome } = this.state;
    return calculateTax(parsedIncome);
  }

  handleIncomeChange = (evt: ChangeEvent<HTMLInputElement>) => {
    this.parseAndSetIncome(evt.target.value);
  };

  parseAndSetIncome = (newIncome: string) => {
    let parsedIncome;
    const cleanIncome = newIncome.replace(/[$,]*/g, '');
    try {
      parsedIncome = Number.parseFloat(mexp.eval(cleanIncome));
    } catch (e) {
      this.setState({
        income: cleanIncome
      });
      return;
    }

    this.setState({
      income: cleanIncome,
      parsedIncome
    });
  };

  handleRemember = () => {
    this.setState(prevState => ({
      rememberedFortnightlyAfterTax:
        (prevState.parsedIncome - this.calculateTax()) / 26
    }));
  };

  handleIncomeTemplate = (template: string) => () => {
    if (this.state.rememberedFortnightlyAfterTax < 0) {
      this.handleRemember();
    }
    this.parseAndSetIncome(template);
  };

  formatCurrency(amount: number) {
    return Number(amount).toLocaleString('en-AU', {
      currency: 'AUD',
      style: 'currency'
    });
  }

  render() {
    const { income, parsedIncome, rememberedFortnightlyAfterTax } = this.state;
    const annualAfterTax = parsedIncome - this.calculateTax();
    return (
      <Container>
        <Card>
          <h1>Quick Tax Calc</h1>
          <p>
            This is a quick calculator for the{' '}
            <a href="https://www.ato.gov.au/Rates/Individual-income-tax-rates/">
              ATO Australian Individual Tax Rates
            </a>
          </p>
          <h2 style={{ marginBottom: 0 }}>How to use</h2>
          <p>
            Type your annual pre-tax amount in the income section and see your
            after tax annual and fortnightly amounts.
          </p>
          <p>
            You can use the memory button (M) to remember a given fortnightly
            amount for comparison.
          </p>
          <p>
            The income section also allows for formulae so you can do things
            like:
          </p>
          <ul>
            <li>
              How might a 10% raise affect my income?{' '}
              <button onClick={this.handleIncomeTemplate(`90000 * 1.1`)}>
                Try it
              </button>
            </li>
            <li>
              What if I get an extra job paying 500 a week?{' '}
              <button onClick={this.handleIncomeTemplate(`90000 + 500 * 52`)}>
                Try it
              </button>
            </li>
            <li>
              If I go 3 days a week, how much will I get paid?{' '}
              <button
                onClick={this.handleIncomeTemplate(`90000 * 3 / 5`)}
                data-testid="template-part-time"
              >
                Try it
              </button>
            </li>
            <li>
              What contractor daily rate is{' '}
              <span
                style={{ textDecoration: 'underline' }}
                title="4 weeks holiday, 2 weeks sick, 10 public holidays, 9.5% super"
              >
                comparable
              </span>{' '}
              with my full time salary?{' '}
              <button
                data-testid="template-contractor"
                onClick={this.handleIncomeTemplate(
                  `(448 * 5 * (52 - 4 - 2 - 2)) / 1.095`
                )}
              >
                Try it
              </button>
            </li>
          </ul>
          <div
            style={{
              backgroundColor: '#eee',
              margin: '0 -3rem',
              padding: '1rem 0.9rem 2rem 3rem'
            }}
          >
            <Row style={{ margin: '1rem 0 2rem' }}>
              <FormLabel htmlFor="income">Income:</FormLabel>
              <Value>
                <input
                  id="income"
                  type="text"
                  value={income}
                  onChange={this.handleIncomeChange}
                  style={{
                    fontSize: '1.4em',
                    textAlign: 'right',
                    border: '1px solid #ccc',
                    maxWidth: '215px'
                  }}
                  data-testid="income"
                />
                <small style={{ display: 'block', textAlign: 'right' }}>
                  Accepts formulas - eg 90000 * 1.1
                </small>
              </Value>
            </Row>
            <Row>
              <Label>Income:</Label>
              <Value>{this.formatCurrency(parsedIncome)}</Value>
            </Row>
            <Row>
              <Label>Tax:</Label>
              <Value>{this.formatCurrency(this.calculateTax())}</Value>
            </Row>
            <Row style={{ marginBottom: '2rem' }}>
              <Label>Annual Less Tax:</Label>
              <Value>{this.formatCurrency(annualAfterTax)}</Value>
            </Row>
            <Row>
              <Label>Fortnightly:</Label>
              <Value>{this.formatCurrency(parsedIncome / 26)}</Value>
            </Row>
            <Row>
              <Label>Fortnightly Tax:</Label>
              <Value>{this.formatCurrency(this.calculateTax() / 26)}</Value>
            </Row>
            <Row style={{ marginTop: '2rem' }}>
              <Label>Fortnightly Less Tax:</Label>
              <Value
                style={{
                  fontSize: '1.5em',
                  marginRight: 0
                }}
              >
                <span data-testid="fortnightly-less-tax">
                  {this.formatCurrency(annualAfterTax / 26)}
                </span>
                <button
                  onClick={this.handleRemember}
                  style={{
                    height: '1.5rem',
                    verticalAlign: 'bottom',
                    marginLeft: '0.5rem'
                  }}
                >
                  M
                </button>
              </Value>
            </Row>
            {rememberedFortnightlyAfterTax >= 0 ? (
              <Fragment>
                <Row>
                  <Label style={{ lineHeight: '2.5em' }}>Remembered:</Label>
                  <Value style={{ fontSize: '1.5em', lineHeight: '1.5em' }}>
                    {this.formatCurrency(rememberedFortnightlyAfterTax)}
                  </Value>
                </Row>
                <Row>
                  <Label>Fortnightly Difference:</Label>
                  <Value
                    style={{
                      fontSize: '1.5em',
                      color:
                        annualAfterTax / 26 - rememberedFortnightlyAfterTax < 0
                          ? '#f00'
                          : '#009e00'
                    }}
                  >
                    {this.formatCurrency(
                      annualAfterTax / 26 - rememberedFortnightlyAfterTax
                    )}
                  </Value>
                </Row>
              </Fragment>
            ) : null}
          </div>
          <p style={{ marginTop: '2rem', textAlign: 'center' }}>
            <a href="https://github.com/zorfling/quick-tax-calc">
              Check out the code on GitHub
            </a>
          </p>
          <p style={{ marginTop: '2rem', textAlign: 'center' }}>
            <a href="https://chriscolborne.com">Built by @zorfling</a>
          </p>
        </Card>
      </Container>
    );
  }
}

export default App;
