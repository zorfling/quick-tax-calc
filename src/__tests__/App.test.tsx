import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import App from '../App';

describe('App', () => {
  test('renders without crashing', () => {
    const container = render(<App />);
    expect(container).toMatchSnapshot();
  });

  test.each([
    ['90000', '$2,661.65'],
    ['120000', '$3,388.58'],
    ['20000', '$756.08'],
    ['185000', '$4,948.19'],
    ['18000', '$692.31']
  ])('calculates basic rates (%i)', (income, expected) => {
    const { getByTestId } = render(<App />);
    const incomeField = getByTestId('income');
    const fortnightlyField = getByTestId('fortnightly-less-tax');

    fireEvent.change(incomeField, { target: { value: income } });
    expect(fortnightlyField).toHaveTextContent(expected);
  });

  test('should work with templates', () => {
    const { getByTestId } = render(<App />);
    const contractorTemplateButton = getByTestId('template-contractor');
    const partTimeTemplateButton = getByTestId('template-part-time');
    const incomeField = getByTestId('income');
    const fortnightlyField = getByTestId('fortnightly-less-tax');

    expect(incomeField).toHaveValue('90000');
    expect(fortnightlyField).toHaveTextContent('$2,661.65');

    fireEvent.click(contractorTemplateButton);
    expect(incomeField).toHaveValue('(448 * 5 * (52 - 4 - 2 - 2)) / 1.095');
    expect(fortnightlyField).toHaveTextContent('$2,661.88');

    fireEvent.click(partTimeTemplateButton);
    expect(incomeField).toHaveValue('90000 * 3 / 5');
    expect(fortnightlyField).toHaveTextContent('$1,727.04');
  });
});
