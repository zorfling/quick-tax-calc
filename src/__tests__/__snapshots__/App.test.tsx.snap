// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdVaJa dsjtQI"
      >
        <div
          class="sc-bwzfXH eRYMzF"
        >
          <h1>
            Quick Tax Calc
          </h1>
          <p>
            This is a quick calculator for the
             
            <a
              href="https://www.ato.gov.au/Rates/Individual-income-tax-rates/"
            >
              ATO Australian Individual Tax Rates
            </a>
          </p>
          <h2
            style="margin-bottom: 0px;"
          >
            How to use
          </h2>
          <p>
            Type your annual pre-tax amount in the income section and see your after tax annual and fortnightly amounts.
          </p>
          <p>
            You can use the memory button (M) to remember a given fortnightly amount for comparison.
          </p>
          <p>
            The income section also allows for formulae so you can do things like:
          </p>
          <ul>
            <li>
              How might a 10% raise affect my income?
               
              <button>
                Try it
              </button>
            </li>
            <li>
              What if I get an extra job paying 500 a week?
               
              <button>
                Try it
              </button>
            </li>
            <li>
              If I go 3 days a week, how much will I get paid?
               
              <button
                data-testid="template-part-time"
              >
                Try it
              </button>
            </li>
            <li>
              What contractor daily rate is
               
              <span
                style="text-decoration: underline;"
                title="4 weeks holiday, 2 weeks sick, 10 public holidays, 9.5% super"
              >
                comparable
              </span>
               
              with my full time salary?
               
              <button
                data-testid="template-contractor"
              >
                Try it
              </button>
            </li>
          </ul>
          <div
            style="background-color: rgb(238, 238, 238); margin: 0px -3rem; padding: 1rem 0.9rem 2rem 3rem;"
          >
            <div
              class="sc-htpNat hqmgw"
            >
              <div
                class="sc-bxivhb kKGLjl"
              >
                Tax Year:
              </div>
              <div
                class="sc-EHOje lclkHR"
              >
                <label>
                  <input
                    type="radio"
                    value="2018"
                  />
                  2017 / 2018
                   
                </label>
                <label>
                  <input
                    type="radio"
                    value="2019"
                  />
                  2018 / 2019
                   
                </label>
                <label>
                  <input
                    checked=""
                    type="radio"
                    value="2020"
                  />
                  2019 / 2020
                   
                </label>
              </div>
            </div>
            <div
              class="sc-htpNat hqmgw"
              style="margin: 1rem 0px 2rem;"
            >
              <label
                class="sc-ifAKCX jhSUeN"
                for="income"
              >
                Income:
              </label>
              <div
                class="sc-EHOje lclkHR"
              >
                <input
                  data-testid="income"
                  id="income"
                  style="font-size: 1.4em; text-align: right; border: 1px solid #ccc; max-width: 215px;"
                  type="text"
                  value="90000"
                />
                <small
                  style="display: block; text-align: right;"
                >
                  Accepts formulas - eg 90000 * 1.1
                </small>
              </div>
            </div>
            <div
              class="sc-htpNat hqmgw"
            >
              <div
                class="sc-bxivhb kKGLjl"
              >
                Income:
              </div>
              <div
                class="sc-EHOje lclkHR"
              >
                A$90,000.00
              </div>
            </div>
            <div
              class="sc-htpNat hqmgw"
            >
              <div
                class="sc-bxivhb kKGLjl"
              >
                Tax:
              </div>
              <div
                class="sc-EHOje lclkHR"
              >
                A$20,797.00
              </div>
            </div>
            <div
              class="sc-htpNat hqmgw"
              style="margin-bottom: 2rem;"
            >
              <div
                class="sc-bxivhb kKGLjl"
              >
                Annual Less Tax:
              </div>
              <div
                class="sc-EHOje lclkHR"
              >
                A$69,203.00
              </div>
            </div>
            <div
              class="sc-htpNat hqmgw"
            >
              <div
                class="sc-bxivhb kKGLjl"
              >
                Fortnightly:
              </div>
              <div
                class="sc-EHOje lclkHR"
              >
                A$3,461.54
              </div>
            </div>
            <div
              class="sc-htpNat hqmgw"
            >
              <div
                class="sc-bxivhb kKGLjl"
              >
                Fortnightly Tax:
              </div>
              <div
                class="sc-EHOje lclkHR"
              >
                A$799.88
              </div>
            </div>
            <div
              class="sc-htpNat hqmgw"
              style="margin-top: 2rem;"
            >
              <div
                class="sc-bxivhb kKGLjl"
              >
                Fortnightly Less Tax:
              </div>
              <div
                class="sc-EHOje lclkHR"
                data-testid="fortnightly-less-tax"
                style="font-size: 1.5em; margin-right: 0px;"
              >
                A$2,661.65
                <button
                  style="height: 1.5rem; vertical-align: bottom; margin-left: 0.5rem;"
                >
                  M
                </button>
              </div>
            </div>
          </div>
          <p
            style="margin-top: 2rem; text-align: center;"
          >
            <a
              href="https://github.com/zorfling/quick-tax-calc"
            >
              Check out the code on GitHub
            </a>
          </p>
          <p
            style="margin-top: 2rem; text-align: center;"
          >
            <a
              href="https://chriscolborne.com"
            >
              Built by @zorfling
            </a>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdVaJa dsjtQI"
    >
      <div
        class="sc-bwzfXH eRYMzF"
      >
        <h1>
          Quick Tax Calc
        </h1>
        <p>
          This is a quick calculator for the
           
          <a
            href="https://www.ato.gov.au/Rates/Individual-income-tax-rates/"
          >
            ATO Australian Individual Tax Rates
          </a>
        </p>
        <h2
          style="margin-bottom: 0px;"
        >
          How to use
        </h2>
        <p>
          Type your annual pre-tax amount in the income section and see your after tax annual and fortnightly amounts.
        </p>
        <p>
          You can use the memory button (M) to remember a given fortnightly amount for comparison.
        </p>
        <p>
          The income section also allows for formulae so you can do things like:
        </p>
        <ul>
          <li>
            How might a 10% raise affect my income?
             
            <button>
              Try it
            </button>
          </li>
          <li>
            What if I get an extra job paying 500 a week?
             
            <button>
              Try it
            </button>
          </li>
          <li>
            If I go 3 days a week, how much will I get paid?
             
            <button
              data-testid="template-part-time"
            >
              Try it
            </button>
          </li>
          <li>
            What contractor daily rate is
             
            <span
              style="text-decoration: underline;"
              title="4 weeks holiday, 2 weeks sick, 10 public holidays, 9.5% super"
            >
              comparable
            </span>
             
            with my full time salary?
             
            <button
              data-testid="template-contractor"
            >
              Try it
            </button>
          </li>
        </ul>
        <div
          style="background-color: rgb(238, 238, 238); margin: 0px -3rem; padding: 1rem 0.9rem 2rem 3rem;"
        >
          <div
            class="sc-htpNat hqmgw"
          >
            <div
              class="sc-bxivhb kKGLjl"
            >
              Tax Year:
            </div>
            <div
              class="sc-EHOje lclkHR"
            >
              <label>
                <input
                  type="radio"
                  value="2018"
                />
                2017 / 2018
                 
              </label>
              <label>
                <input
                  type="radio"
                  value="2019"
                />
                2018 / 2019
                 
              </label>
              <label>
                <input
                  checked=""
                  type="radio"
                  value="2020"
                />
                2019 / 2020
                 
              </label>
            </div>
          </div>
          <div
            class="sc-htpNat hqmgw"
            style="margin: 1rem 0px 2rem;"
          >
            <label
              class="sc-ifAKCX jhSUeN"
              for="income"
            >
              Income:
            </label>
            <div
              class="sc-EHOje lclkHR"
            >
              <input
                data-testid="income"
                id="income"
                style="font-size: 1.4em; text-align: right; border: 1px solid #ccc; max-width: 215px;"
                type="text"
                value="90000"
              />
              <small
                style="display: block; text-align: right;"
              >
                Accepts formulas - eg 90000 * 1.1
              </small>
            </div>
          </div>
          <div
            class="sc-htpNat hqmgw"
          >
            <div
              class="sc-bxivhb kKGLjl"
            >
              Income:
            </div>
            <div
              class="sc-EHOje lclkHR"
            >
              A$90,000.00
            </div>
          </div>
          <div
            class="sc-htpNat hqmgw"
          >
            <div
              class="sc-bxivhb kKGLjl"
            >
              Tax:
            </div>
            <div
              class="sc-EHOje lclkHR"
            >
              A$20,797.00
            </div>
          </div>
          <div
            class="sc-htpNat hqmgw"
            style="margin-bottom: 2rem;"
          >
            <div
              class="sc-bxivhb kKGLjl"
            >
              Annual Less Tax:
            </div>
            <div
              class="sc-EHOje lclkHR"
            >
              A$69,203.00
            </div>
          </div>
          <div
            class="sc-htpNat hqmgw"
          >
            <div
              class="sc-bxivhb kKGLjl"
            >
              Fortnightly:
            </div>
            <div
              class="sc-EHOje lclkHR"
            >
              A$3,461.54
            </div>
          </div>
          <div
            class="sc-htpNat hqmgw"
          >
            <div
              class="sc-bxivhb kKGLjl"
            >
              Fortnightly Tax:
            </div>
            <div
              class="sc-EHOje lclkHR"
            >
              A$799.88
            </div>
          </div>
          <div
            class="sc-htpNat hqmgw"
            style="margin-top: 2rem;"
          >
            <div
              class="sc-bxivhb kKGLjl"
            >
              Fortnightly Less Tax:
            </div>
            <div
              class="sc-EHOje lclkHR"
              data-testid="fortnightly-less-tax"
              style="font-size: 1.5em; margin-right: 0px;"
            >
              A$2,661.65
              <button
                style="height: 1.5rem; vertical-align: bottom; margin-left: 0.5rem;"
              >
                M
              </button>
            </div>
          </div>
        </div>
        <p
          style="margin-top: 2rem; text-align: center;"
        >
          <a
            href="https://github.com/zorfling/quick-tax-calc"
          >
            Check out the code on GitHub
          </a>
        </p>
        <p
          style="margin-top: 2rem; text-align: center;"
        >
          <a
            href="https://chriscolborne.com"
          >
            Built by @zorfling
          </a>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
